deepcell_tracking/utils.py

```diff
@@ -203,23 +206,88 @@ def save_trks(filename, lineages, raw, tracked):
     """Saves raw, tracked, and lineage data into one trks_file.
 
     Args:
-        filename (str): full path to the final trk files.
-        lineages (dict): a list of dictionaries saved as a json.
+        filename (str or io.BytesIO): full path to the final trk files or bytes object
+            to save the data to
+        lineages (list): a list of dictionaries saved as a json.
         raw (np.array): raw images data.
         tracked (np.array): annotated image data.
 
+    Returns:
+        io.BytesIO: If filename is instance of io.BytesIO
+
     Raises:
         ValueError: filename does not end in ".trks".
     """
-    if not str(filename).lower().endswith('.trks'):
+    ext = os.path.splitext(str(filename))[-1]
+    if not isinstance(filename, io.BytesIO) and ext != '.trks':
         raise ValueError('filename must end with `.trks`. Found %s' % filename)
 
-    with tarfile.open(filename, 'w:gz') as trks:
+    save_track_data(filename=filename,
+                    lineages=lineages,
+                    raw=raw,
+                    tracked=tracked,
+                    lineage_name='lineages.json')
+
+
+def save_trk(filename, lineages, raw, tracked):
+    """Saves raw, tracked, and lineage data into a trk_file.
+
+    Args:
+        filename (str or io.BytesIO): full path to the final trk files or bytes object
+            to save the data to
+        lineages (list or dict): a list of a single dictionary or a single lineage dictionarys
```

Comment:
This line looks kind of long, and I think this is misspelled:

```suggestion
        lineages (list or dict): a list of a single dictionary or a single lineage dictionaries
```